freqtrade/exchange/__init__.py

```diff
@@ -333,6 +339,59 @@ def sell(self, pair: str, ordertype: str, amount: float, rate: float) -> Dict:
         except ccxt.BaseError as e:
             raise OperationalException(e)
 
+    def stoploss_limit(self, pair: str, amount: float, stop_price: float, rate: float) -> Dict:
+        """
+        creates a stoploss limit order.
+        NOTICE: it is not supported by all exchanges. only binance is tested for now.
+        """
+
+        # Set the precision for amount and price(rate) as accepted by the exchange
+        amount = self.symbol_amount_prec(pair, amount)
+        rate = self.symbol_price_prec(pair, rate)
+        stop_price = self.symbol_price_prec(pair, stop_price)
+
+        # Ensure rate is less than stop price
+        if stop_price <= rate:
+            raise OperationalException(
+                'In stoploss limit order, stop price should be more than limit price')
+
+        if self._conf['dry_run']:
+            order_id = f'dry_run_buy_{randint(0, 10**6)}'
+            self._dry_run_open_orders[order_id] = {
+                'info': {},
+                'id': order_id,
+                'pair': pair,
+                'price': stop_price,
+                'amount': amount,
+                'type': 'stop_loss_limit',
+                'side': 'sell',
+                'remaining': amount,
+                'datetime': arrow.utcnow().isoformat(),
+                'status': 'open',
+                'fee': None
+            }
+            return self._dry_run_open_orders[order_id]
+
+        try:
+            return self._api.create_order(pair, 'stop_loss', 'sell',
+                                          amount, rate, {'stopPrice': stop_price})
+
+        except ccxt.InsufficientFunds as e:
+            raise DependencyException(
+                f'Insufficient funds to place stoploss limit order on market {pair}.'
+                f'Tried to put a stoploss amount {amount} at rate {rate} (total {rate*amount}).'
+                f'Message: {e}')
+        except ccxt.InvalidOrder as e:
+            raise DependencyException(
+                f'Could not place stoploss limit order on market {pair}.'
+                f'Tried to place stoploss amount {amount} at rate {rate} (total {rate*amount}).'
+                f'Message: {e}')
+        except (ccxt.NetworkError, ccxt.ExchangeError) as e:
+            raise TemporaryError(
+                f'Could not place stoploss limit order due to {e.__class__.__name__}. Message: {e}')
+        except ccxt.BaseError as e:
```

Comment:
i would want some tests for this ... makes me sleep better as it avoids errors such as the one this morning, just because this code never ran ...

Reply:
you mean real test not unit one right ?

Reply:
well, the problem of this morning was the wallet which is used globally by the bot. but this is just a feature which is off by default. 
I was going to test it on live once it is merged.

Reply:
best take  `test_sell_prod` as a sample - it contains tests for the different exception handlers - making sure that the different exceptions ccxt can throw at us are cought and changed to one of "our" exceptions - so we have ccxt exceptionhandling focused in one place.

Reply:
Ok I am going to add it. don't be too picky on this one. anyway I think once it is running live we will have more surprises. as long as it doesn't impact other parts (it is off by default) we should be able to merge and test it. I am not against adding a notice on documentation which explains that this feature is a work in progress.